Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -155,24 +155,26 @@ struct ComponentSpec {
 
 struct StartOfFrame {
 
-    // Of these, only the first 3 are in mainstream use, and refers to SOF0-2.
-    enum class FrameType {
-        Baseline_DCT = 0,
-        Extended_Sequential_DCT = 1,
-        Progressive_DCT = 2,
-        Sequential_Lossless = 3,
-        Differential_Sequential_DCT = 5,
-        Differential_Progressive_DCT = 6,
-        Differential_Sequential_Lossless = 7,
-        Extended_Sequential_DCT_Arithmetic = 9,
-        Progressive_DCT_Arithmetic = 10,
-        Sequential_Lossless_Arithmetic = 11,
-        Differential_Sequential_DCT_Arithmetic = 13,
-        Differential_Progressive_DCT_Arithmetic = 14,
-        Differential_Sequential_Lossless_Arithmetic = 15,
+    // Only the combinations (Huffman, Baseline_DCT, false), (Huffman, Sequential_DCT, false), (Huffman, Progressive_DCT, false) are in mainstream use
+    struct FrameType {
+        enum class Encoding {
+            Huffman = 0,
+            Arithmetic = 1,
+        };
+
+        enum class Type {
+            Baseline_DCT = 0,
+            Sequential_DCT = 1,
+            Progressive_DCT = 2,
+            Lossless = 3,
+        };
+
+        Encoding encoding { Encoding::Huffman };
+        Type type { Type::Baseline_DCT };
+        bool differential { false };
```

Comment:
The name `FrameType` doesn't suit the struct as it clearly includes more information about the frame than just the type. Also, `frame.type.type` looks weird. Maybe rename the struct and the variable `type` to something else? Or alternatively, leave `FrameType` an enum and move non-type related data outside?